core/src/main/kotlin/com/bumble/appyx/core/node/ParentNodeExt.kt

```diff
@@ -0,0 +1,15 @@
+package com.bumble.appyx.core.node
+
+import com.bumble.appyx.core.children.nodeOrNull
+
+fun ParentNode<*>.children(): List<Node> {
+    return this.children.value.values.mapNotNull { it.nodeOrNull }
+}
+
+inline fun <reified N : Node> ParentNode<*>.childrenOfType(): List<N> {
+    return this.children.value.values.mapNotNull { it.nodeOrNull }.filterIsInstance<N>()
+}
+
+inline fun <reified N : Node> ParentNode<*>.firstChildOfType(): N {
+    return childrenOfType<N>().first()
```

Comment:
Here we can avoid list allocation by using `firstOrNull { it.nodeOrNull as? N }`

Not sure, should we crash or not in case of missing child? Maybe make nullable?